File: modelo/medico.py
from modelo.persona import Persona


class Medico(Persona):
    def __init__(self, dni: str, nombre: str, edad: int, cmp: str, especialidad: str):
        super().__init__(dni, nombre, edad)

        if not cmp or not cmp.strip():
            raise ValueError("El CMP es obligatorio")

        if not especialidad or not especialidad.strip():
            raise ValueError("La especialidad es obligatoria")

        self._cmp = cmp.strip()
        self._especialidad = especialidad.strip()

    @property
    def rol(self) -> str:
        return "Médico"

    @property
    def cmp(self) -> str:
        return self._cmp

    @property
    def especialidad(self) -> str:
        return self._especialidad

    def mostrar_datos(self) -> str:
        return (
            super().mostrar_datos()
            + f"\nCMP: {self._cmp}"
            + f"\nEspecialidad: {self._especialidad}"
        )

File: src/centrosalud/Medico.java
package centrosalud;

public class Medico extends Persona {
    private String cmp;
    private String especialidad;

    public Medico(String dni, String nombres, String apellidos, String fechaNacimiento,
                   String cmp, String especialidad) {
        super(dni, nombres, apellidos, fechaNacimiento);
        this.cmp = cmp;
        this.especialidad = especialidad;
    }

    public String getCmp() {
        return cmp;
    }

    public String getEspecialidad() {
        return especialidad;
    }

    public void atenderCita(CitaMedica cita) {
        System.out.println("El médico " + getNombreCompleto()
                + " está atendiendo la cita " + cita.getIdCita());
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("CMP: " + cmp);
        System.out.println("Especialidad: " + especialidad);
    }
}